raspberry/bzzz/sensors/evo_time_of_flight.py

```diff
@@ -0,0 +1,250 @@
+        :param log_file: file name to log data; default: None
+        :param max_samples: maximum number of samples to record; default: 100000
+ 
+        If `log_file` is None, the data is not logged; otherwise, on exit,
+        the data are stored in a CSV file
+ 
+        Note: We assume that we receive 1 distance measurement from the Evo time of flight sensor
+        """
+        # Open the correct port
+        self.__openEvo
+        # A lock is used to guarantee that we won't be reading the data
+        # while the thread in the background is writing it
+        self.__lock = Lock()
+        self.__thread = Thread(target=self.__get_measurements_in_background_t,
+                               args=[serial_path, baud])
+        self.__keep_going = True
+        self.__window_length = window_length
+        self.__values_cache = np.tile(np.nan, (self.__window_length, 1))
+        self.__cursor = 0
+        self.__data_processor = data_processor
+        self.__log_file = log_file
+        self.__max_samples = max_samples
+        if log_file is not None:
+            feature_names = ("Date_Time","Altitude")
+            self.__logger = DataLogger(num_features=1,
+                                       max_samples=max_samples,
+                                       feature_names=feature_names)
+        self.__thread.start()
+ 
+   
+    def __openEvo(self, serial_path, baud):
+        # Open the Evo and catch any exceptions thrown by the OS
+        evo = serial.Serial(serial_path, baud, timeout=2)
+        # Send the command "Binary mode"
+        set_bin = (0x00, 0x11, 0x02, 0x4C)
+        # Flush in the buffer
+        evo.flushInput()
+        # Write the binary command to the Evo
+        evo.write(set_bin)
+        # Flush out the buffer
+        evo.flushOutput()
+        return #evo
+   
+ 
+    def __get_measurements_in_background_t(self, serial_path, baud):
+        """
+        This is a thread that runs in the background to connect to the
+        serial and collect measurements, which are stored in a buffer
+ 
+        :param serial_path: serial path
+        :param baud: baud rate of serial communication
+        """
+        crc8_fn = crcmod.predefined.mkPredefinedCrcFun('crc-8')
+        ser = serial.Serial(serial_path, baud, timeout=1)
+        ser.reset_input_buffer()
+        while True:
+            if ser.in_waiting > 0:
+                sensor_data = ser.read(1)
+                if sensor_data == b'T':
+                    # After T read 3 bytes
+                    frame = sensor_data + ser.read(3)
+                    if frame[3] == crc8_fn(frame[0:3]):
+                        # Convert binary frame to decimal in shifting by 8 the frame
+                        rng = frame[1] << 8
+                        rng = rng | (frame[2] & 0xFF)
+                    else:
+                        return "CRC mismatch. Check connection or make sure only one progam access the sensor port."
```

Comment:
If this happes, the function `__get_measurements_in_background_t` will retur and then you stop getting measurements (the thread will terminate). I guess if the CRC doesn't check out we should just ignore the measurement and move on?

Reply:
yea I think we should just ignore it as it updates very quickly